File: .API/Models/UserProfile.cs
// Decompiled with JetBrains decompiler
// Type: CloudX.Shared.UserProfile
// Assembly: CloudX.Shared, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
// MVID: 6223B97A-06A5-46CB-9E10-78604961D6EE
// Assembly location: J:\D\SteamLibrary\steamapps\common\NeosVR\HeadlessClient\CloudX.Shared.dll

using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.Json.Serialization;

namespace CloudX.Shared
{
  [JsonObject(MemberSerialization = MemberSerialization.OptIn)]
  public class UserProfile
  {
    public static int MAX_SHOWCASE_ITEMS
    {
      get
      {
        return 6;
      }
    }

    [JsonProperty(PropertyName = "iconUrl")]
    [JsonPropertyName("iconUrl")]
    public string IconUrl { get; set; }

    [JsonProperty(PropertyName = "backgroundUrl")]
    [JsonPropertyName("backgroundUrl")]
    public string BackgroundUrl { get; set; }

    [JsonProperty(PropertyName = "tagline")]
    [JsonPropertyName("tagline")]
    public string Tagline { get; set; }

    [JsonProperty(PropertyName = "description")]
    [JsonPropertyName("description")]
    public string Description { get; set; }

    [JsonProperty(PropertyName = "profileWorldUrl")]
    [JsonPropertyName("profileWorldUrl")]
    public string ProfileWorldUrl { get; set; }

    [JsonProperty(PropertyName = "showcaseItems")]
    [JsonPropertyName("showcaseItems")]
    public List<string> ShowcaseItems { get; set; }

    [JsonProperty(PropertyName = "tokenOptOut")]
    [JsonPropertyName("tokenOptOut")]
    public List<string> TokenOptOut { get; set; }

    [JsonIgnore]
    [JsonIgnore]
    public bool IsValid
    {
      get
      {
        List<string> showcaseItems = this.ShowcaseItems;
        return (showcaseItems != null ? __nonvirtual (showcaseItems.Count) : 0) <= UserProfile.MAX_SHOWCASE_ITEMS;
      }
    }

    public bool AcceptsToken(string token)
    {
      return this.TokenOptOut == null || !this.TokenOptOut.Any<string>((Func<string, bool>) (s => s.Equals(token, StringComparison.InvariantCultureIgnoreCase)));
    }
  }
}

File: src/api/UserProfile.ts
import { List } from "@bombitmanbomb/utils";
export class UserProfile {
	IconUrl: string;
	BackgroundUrl: string;
	TagLine: string;
	ProfileWorldUrl: string;
	ShowcaseItems: List<string>;
	TokenOptOut: List<string>;
	constructor($b: UserProfileJSON) {
		this.IconUrl = $b.iconUrl;
		this.BackgroundUrl = $b.backgroundUrl;
		this.TagLine = $b.tagLine;
		this.ProfileWorldUrl = $b.profileWorldUrl;
		this.ShowcaseItems =
			$b.showcaseItems instanceof List
				? $b.showcaseItems
				: List.ToList($b.showcaseItems);
		this.TokenOptOut =
			$b.tokenOptOut instanceof List
				? $b.tokenOptOut
				: List.ToList($b.tokenOptOut);
	}
	public static MAX_SHOWCASE_ITEMS = 6;
	public get IsValid(): boolean {
		return (
			(this.ShowcaseItems != null ? this.ShowcaseItems.Count : 0) <=
			UserProfile.MAX_SHOWCASE_ITEMS
		);
	}
	public AcceptsToken(token: string): boolean {
		return (
			this.TokenOptOut.length == 0 || !this.TokenOptOut.some((s) => token == s)
		);
	}
	toJSON(): UserProfileJSON {
		return {
			iconUrl: this.IconUrl,
			backgroundUrl: this.BackgroundUrl,
			tagLine: this.TagLine,
			profileWorldUrl: this.ProfileWorldUrl,
			showcaseItems: this.ShowcaseItems,
			tokenOptOut: this.TokenOptOut,
		};
	}
}
export interface UserProfileJSON {
	iconUrl: string;
	backgroundUrl: string;
	tagLine: string;
	profileWorldUrl: string;
	showcaseItems: string[];
	tokenOptOut: string[];
}
